extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js

```diff
@@ -24,8 +26,13 @@ const FONTS_TO_LOAD = [
   },
 ];
 
-const renderShoppingTagTemplate = (tagData) => (
-  <div class="amp-story-shopping-tag-inner">
+/**
+ * @param {!ShoppingConfigDataDef} tagData
+ * @param {function(Event)} onClick
+ * @return {!Element}
+ */
+const renderShoppingTagTemplate = (tagData, onClick) => (
+  <div class="amp-story-shopping-tag-inner" role="button" onClick={onClick}>
```

Comment:
Please see comment on line 84. 
We can send `tagData` through the `onClick` so that we don't have to get data from the store service. 
The onClick could be something like:
`onClick={() => onClick(tagData)}`

Reply:
Ok this works, but I'm just curious why doesn't just using the tagData passed in as a parameter work, why do we need to "bind" it twice?

Reply:
This is not a matter of scoping/binding. This is a function expression.

Writing `onClick(tagData)` will call the function when the template is being built.

We need a function expression `() => onClick(tagData)` for it to run onClick.